code/modules/surgery/organs/eyes.dm

```diff
@@ -79,3 +112,78 @@
 	desc = "An electronic device designed to mimic the functions of a pair of human eyes. It has no benefits over organic eyes, but is easy to produce."
 	origin_tech = "biotech=4"
 	status = ORGAN_ROBOT
+
+/obj/item/organ/internal/eyes/cybernetic/emp_act(severity)
+	if(!owner || emp_proof)
+		return
+	if(prob(10 * severity))
+		return
+	to_chat(owner, "<span class='warning'>Static obfuscates your vision!</span>")
+	owner.flash_eyes(visual = TRUE)
+	..()
+
+/obj/item/organ/internal/eyes/cybernetic/meson
+	name = "meson eyes"
+	desc = "These cybernetic eyes will allow you to see the structural layout of the station, and, well, everything else."
+	eye_color = "#199900"
+	vision_flags = SEE_TURFS
+	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE
+	origin_tech = "materials=4;engineering=4;biotech=4;magnets=4"
+
+/obj/item/organ/internal/eyes/cybernetic/xray
+	name = "\improper X-ray eyes"
+	desc = "These cybernetic eyes will give you X-ray vision. Blinking is futile."
+	see_in_dark = 8
+	vision_flags = SEE_MOBS | SEE_OBJS | SEE_TURFS
+	origin_tech = "materials=4;programming=4;biotech=7;magnets=4"
+
+/obj/item/organ/internal/eyes/cybernetic/thermals
+	name = "thermal eyes"
+	desc = "These cybernetic eye implants will give you thermal vision. Vertical slit pupil included."
+	eye_color = "#FFCC00"
+	vision_flags = SEE_MOBS
+	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE
+	flash_protect = FLASH_PROTECTION_SENSITIVE
+	see_in_dark = 8
+	origin_tech = "materials=5;programming=4;biotech=4;magnets=4;syndicate=1"
+
+/obj/item/organ/internal/eyes/cybernetic/flashlight
+	name = "flashlight eyes"
+	desc = "It's two flashlights rigged together with some wire. Why would you put these in someone's head?"
+	eye_color ="#FEE5A3"
+	icon = 'icons/obj/lighting.dmi'
+	icon_state = "flashlight_eyes"
+	flash_protect = FLASH_PROTECTION_WELDER
+	tint = INFINITY
+	var/obj/item/flashlight/eyelight/eye
+
+/obj/item/organ/internal/eyes/cybernetic/flashlight/emp_act(severity)
+	return
+
+/obj/item/organ/internal/eyes/cybernetic/flashlight/insert(mob/living/carbon/M, special = FALSE)
+	..()
+	if(!eye)
+		eye = new /obj/item/flashlight/eyelight()
+	eye.on = TRUE
+	eye.forceMove(M)
+	eye.update_brightness(M)
+	M.BecomeBlind()
+
+
+/obj/item/organ/internal/eyes/cybernetic/flashlight/remove(mob/living/carbon/M, special = FALSE)
+	eye.on = FALSE
+	eye.update_brightness(M)
+	eye.forceMove(src)
+	M.CureBlind()
+	. = ..()
```

Comment:
Doesn't actually make any difference here, but more often than not `. = ..()` is at the top of the proc.
```suggestion
	. = ..()
	eye.on = FALSE
	eye.update_brightness(M)
	eye.forceMove(src)
	M.CureBlind()
```